src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -2882,110 +712,4 @@ internal static BinaryOperationKind DeriveBinaryOperationKind(BinaryOperatorKind
-    {
-        public override void Accept(OperationVisitor visitor)
-        {
-            throw ExceptionUtilities.Unreachable;
-        }
-
-        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            throw ExceptionUtilities.Unreachable;
-        }
-
-        protected override OperationKind ExpressionKind
-        {
-            get { throw ExceptionUtilities.Unreachable; }
-        }
-    }
-
-    partial class BoundIsPatternExpression
-    {
-        public override void Accept(OperationVisitor visitor)
-        {
-            // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-            visitor.VisitNoneOperation(this);
-        }
-
-        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-            return visitor.VisitNoneOperation(this, argument);
-        }
-
-        // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-        protected override OperationKind ExpressionKind => OperationKind.None;
-    }
-
-    /// <summary>
-    /// This node represents an out or deconstruction variable.
-    /// It is only used temporarily during initial binding.
-    /// </summary>
-    internal partial class VariablePendingInference
-    {
-        public override void Accept(OperationVisitor visitor)
-        {
-            throw ExceptionUtilities.Unreachable;
-        }
-
-        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            throw ExceptionUtilities.Unreachable;
-        }
-
-        protected override OperationKind ExpressionKind
-        {
-            get { throw ExceptionUtilities.Unreachable; }
-        }
-    }
-
-    partial class BoundThrowExpression
-    {
-        public override void Accept(OperationVisitor visitor)
-        {
-            // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-            visitor.VisitNoneOperation(this);
-        }
-
-        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-            return visitor.VisitNoneOperation(this, argument);
-        }
-
-        // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
-        protected override OperationKind ExpressionKind => OperationKind.None;
-    }
-
-    internal partial class BoundDeclarationPattern
```

Comment:
>internal partial class BoundDeclarationPattern [](start = 4, length = 46)

This partial declaration should be moved to Constructors.cs rather than deleted.